argparse_dataclass.py

```diff
@@ -186,3 +186,14 @@ def parse_args(self, *args, **kwargs) -> OptionsType:
         """Parse arguments and return as the dataclass type."""
         namespace = super().parse_args(*args, **kwargs)
         return self._options_type(**vars(namespace))
+
+
+def dataclass(cls):
```

Comment:
For this to be viable it would need to take the same optional arguments as the upstream `dataclass` decorator.